src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -267,6 +290,16 @@ private static IEnumerable<T> ReadFromProtobufStream<T>(Stream stream, MessagePa
             return new Result<T>(httpResponse, parseStringResult(stringResponse));
         }
 
+        private void ThrowsIfError(JObject parsedJson)
+        {
+            var firstErrorMessage = parsedJson["errors"]?.Select(x => x["msg"]?.ToObject<string>())
+                .FirstOrDefault();
+            if (firstErrorMessage != null)
+            {
+                throw new Exception(firstErrorMessage);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)

Reply:
Comment for later: we're throwing System.Exception in a number of places - we should either disable the rule or come up with a better solution.

Reply:
I won't do anything for now but clearly my feeling is that we should keep the rule and create custom exceptions.